src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceDescriptor.cs

```diff
@@ -83,6 +106,14 @@ public static ServiceDescriptor Transient([NotNull] Type service, [NotNull] Type
             return Describe(service, implementationType, ServiceLifetime.Transient);
         }
 
+        public static ServiceDescriptor Transient<TService, TImplementation>(
+            [NotNull] Func<IServiceProvider, TImplementation> implementationFactory)
+            where TService : class
+            where TImplementation : class
```

Comment:
I think we should also use stronger generic constraints in this class where possible.